src/jackdaw/serdes/avro.clj

```diff
@@ -525,8 +525,14 @@
                                 (catch clojure.lang.ExceptionInfo e
                                   (let [data (-> e
                                                  ex-data
-                                                 (assoc :topic topic :clj-data data))]
-                                    (throw (ex-info (.getMessage e) data))))))}
+                                                 (assoc :topic topic :clj-data data))
+                                        path (:path data)
+                                        info (str (merge {:message (.getMessage e)
+                                                          :topic   (:topic data)}
+                                                         (when (not (empty? path))
```

Comment:
```clojure
(when-let [path (:path data)]
   {:path path
    :data (get-in (:clj-data data) path)})
```
better cohesion 💁‍♂️

Reply:
Sure can move it into a when-let with a seq. The path is never nil, if not used it's an empty vector.

Reply:
Yeah good catch, so 
```clojure
when-let [path (-> data :path seq)]
    {:path path
     :data (get-in (:clj-data data) path)})
```

Reply:
I made any changes and expanded the PR description. The one thing that I don't like is that for errors such as missing required field or unknown field, the path is pointing to the map said fields are missing from or are found. Which is different to the behaviour for the type errors. So if something is nested, with this code you might end up getting the data for the whole map that contains any attribute with erroneous data. We could just omit the data and always include the path in the message which would be more consistent, or just always include the data under any path (which would make messages very verbose - as most errors are on the top level usually)

Reply:
I went for a slightly different approach. I always include the path in the error message and then the data are only included if it's a type error that was raised (as in any other case we don't really care what the data is)